core/src/system_monitor_service.rs

```diff
@@ -832,6 +836,47 @@ impl SystemMonitorService {
         Self::report_cpuid_values();
     }
 
+    fn get_open_fd_stats() -> Option<(usize, usize, usize, usize)> {
+        let proc = Process::myself().ok()?;
+        let curr_num_open_fd = proc.fd_count().unwrap();
+        let curr_mmap_count = proc.maps().unwrap().len();
```

Comment:
Have you measured how long this takes to execute with 2M open fds?

Reply:
Yeah. It takes about 10.520629985s for 2M open fds. Maybe we can increase the report interval from 30s to 3 minutes?

Reply:
may very well miss an event on a three minute interval

Reply:
This all seems pretty blunt. Beyond append vecs, which seems like it would be covered by https://github.com/solana-labs/solana/pull/28958, what other sources of unbounded fd/mmap growth exist and perhaps we can directly instrument those as well.

Reply:
the dumb `wc -l /proc/PID/maps` is like 600ms (850ms if the file is copied aside first) on a box with 505k maps.  should be linear to 2M, ~2.5s.  i'm guessing it's all the parsing that's of no use to us which is accounting for the rest of that time?

Reply:
```
sol@kin-validator-am6-1:~$ wc -l /proc/30565/maps
282764 /proc/30565/maps
```
![image](https://user-images.githubusercontent.com/219428/204640509-1a68f287-29bf-4038-a690-fa6d1ffc2a48.png)
#28958 shows 87K, while the proc maps shows 282K.
@jeffwashington any idea for the difference?

Reply:
```
$ time cat maps | sed -e 's:.*/\(accounts\|accounts_index\)/.*:\1:;t;c\
other' | sort | uniq -c
  416730 accounts
  82965 accounts_index
  4980 other
```

Reply:
sol@kin-validator-am6-1:~$ cat /proc/30565/maps| sed -e 's:.*/\(accounts\|accounts_index\)/.*:\1:;t;c\other' | sort | uniq -c
  87885 accounts
 191468 accounts_index
   3399 other

Reply:
thanks! I tried @t-nelson's command. Looks like https://github.com/solana-labs/solana/pull/28958 tracks the append-vecs correctly. the other large chucnk of mmap is used for accounts_index, 191K. @jeffwashington probably, we should track the number of accounts_index mmap file too?

Reply:
> the dumb `wc -l /proc/PID/maps` is like 600ms (850ms if the file is copied aside first) on a box with 505k maps. should be linear to 2M, ~2.5s. i'm guessing it's all the parsing that's of no use to us which is accounting for the rest of that time?

yeah. it takes 1.51s to count 2M open mmaps.

Reply:
> thanks! I tried @t-nelson's command. Looks like #28958 tracks the append-vecs _correctly_. the other large chunk of mmaps are used for accounts_index, 191K. @jeffwashington probably, we should track the number of accounts_index mmap file too?

Here is the change to track accounts index mmap files.
https://github.com/solana-labs/solana/pull/28984